WireDomain/Sources/WireDomain/Utilities/Journal/JournalKey.swift

```diff
@@ -69,6 +69,30 @@ public extension JournalKey where Value == Bool {
 
 }
 
+public extension JournalKey where Value == Bool? {
+
+    /// Whether the user has given consent to track analytics.
+
+    static let isAnalyticsTrackingConsentGiven = Self(
+        "isAnalyticsTrackingConsentGiven",
+        defaultValue: nil
+    )
```

Comment:
question: do I understand correctly that if there is consent, then there should be an analytics id, and if there is no consent, there should be no analytics id? If so, perhaps we can avoid this flag and depend on the existence of the id.

Reply:
We need a third option specifying that the user hasn't been asked yet.
Only then we will present the alert on the first login.